app/models/course/assessment/question/programming.rb

```diff
@@ -20,7 +20,7 @@ class Course::Assessment::Question::Programming < ApplicationRecord # rubocop:di
   acts_as :question, class_name: 'Course::Assessment::Question'
 
   after_initialize :set_defaults
-  after_save :create_codaveri_problem, if: :duplicating?
+  after_save :create_or_update_codaveri_problem, if: :duplicating?
```

Comment:
the first commit does not handle duplication on-demand yet, and hence I kept it there. Removing it from the first commit will lead to the duplication behavior at this point to be inconsistent

i.e. when duplicating, the `codaveri_id` will be brought over to the new object as well.

Another option to fulfill this request is to squash the first commit and second commit altogether, though I personally do not prefer this since it will make one commit to be unnecessarily big.